data/rulesets.js

```diff
@@ -372,10 +372,25 @@ exports.BattleFormats = {
 		},
 		onValidateTeam: function (team, format) {
 			let abilityTable = {};
+			let clones = {
+				airlock: ['cloud9'],
+				battlearmor: ['shellarmor'],
+				clearbody: ['whitesmoke'],
+				cloud9: ['airlock'],
+				dazzling: ['queenlymajesty'],
+				moldbreaker: ['teravolt', 'turboblaze'],
+				powerofalchemy: ['receiver'],
+				queenlymajesty: ['dazzling'],
+				receiver: ['powerofalchemy'],
+				shellarmor: ['battlearmor'],
+				teravolt: ['moldbreaker', 'turboblaze'],
+				turboblaze: ['moldbreaker', 'teravolt'],
+				whitesmoke: ['clearbody'],
+			};
 			for (let i = 0; i < team.length; i++) {
 				let ability = toId(team[i].ability);
 				if (!ability) continue;
-				if (ability in abilityTable) {
+				if (ability in abilityTable || clones[ability].some(clone => {return clone in abilityTable;})) {
```

Comment:
This is actually just messed-up logic that I think you have more than one logical error which is why I can't pinpoint it. Make sure you understand what each variable means, and what your loop invariant is.